### PR TITLE
chore: Rename inputs parameter to sources for on demand feature views

### PR DESCRIPTION
**Pull request by felixwang9817**
**What this PR does / why we need it**: This PR renames the `inputs` parameter to `sources` for odfvs.

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

**Comment by felixwang9817** [comment]
there are some conflicts with Go, so I will hold this PR until #2429 is merged

File: sdk/python/feast/on_demand_feature_view.py
Comment (achals) [inline sdk/python/feast/on_demand_feature_view.py:76]: I think making this a `kwarg` from an `arg` is a backwards incompatibile change. e.g.
```
ODFV(
  "my name",
  [...],
  {"input1": some_feature_view}
)
```

will no longer be valid. Are you thinking that this is okay because it's experimental?

Comment (felixwang9817) [reply sdk/python/feast/on_demand_feature_view.py:76]: reordering parameters so that this doesn't break

**Review by achals — APPROVED** [review]
/lgtm

**Pull request merged**
